### PR TITLE
Add sentry reports to src/embed

### LINKED ISSUE #228: Complete sentry reports in src/embed.ts

### PR DESCRIPTION
**Pull request by kulla**
Closes #228

### CODE REVIEW

**Comment by kulla** [comment]
@elbotho Can you make the review for this PR since Caro is currently ill? Thanks!

File: src/embed.ts [hunk 1]
```diff
@@ -94,40 +95,92 @@ async function getVimeoThumbnail(url: URL, sentry: SentryReporter) {
     'https://vimeo.com/api/oembed.json?url=' +
       encodeURIComponent(`https://vimeo.com/${videoId}`)
   )
+  const apiResponseText = await apiResponse.text()
+
+  if (apiResponse.status !== 200) {
+    if (apiResponse.status !== 404) {
```
Comment (elbotho) [inline src/embed.ts:101]: why exclude 404?

File: src/embed.ts [hunk 2]
```diff
@@ -94,40 +95,92 @@ async function getVimeoThumbnail(url: URL, sentry: SentryReporter) {
...
+      'apiResponse',
+      responseToContext({ response: apiResponse, json: apiResponseJson })
+    )
+    sentry.captureMessage('Vimeo API returns unsupported JSON', 'warning')
+    return getPlaceholder()
+  }
 
-    const url = returnedJson.thumbnail_url.replace(/_[0-9|x]+/, '')
+  let imgUrl: Url
+  const vimeoThumbnailUrl = apiResponseJson.thumbnail_url.replace(
```
Comment (elbotho) [inline src/embed.ts:137]: aside: why is this necessary, I don't remember?

**Review by elbotho — APPROVED** [review]
Looks good!
All in all it seems very exact a lot of code,
have there been a lot of problems in this module or do you want to archive this level of reporting everywhere in the long run?

File: src/embed.ts (hunk 1 above)
Comment (kulla) [reply src/embed.ts:101]: Vimeo returns a 404 response when the video does not exist (= the video url is wrong). This case is not interesting for us.

File: src/embed.ts (hunk 2 above)
Comment (kulla) [reply src/embed.ts:137]: I have no idea either :see_no_evil: Might want to use the returned thumbnail url...

File: src/embed.ts (hunk 1 above)
Comment (elbotho) [reply src/embed.ts:101]: Hm. It's not a code problem. But we might log it anyway and pass it on to content ppl (or fix it ourselves) since it's still a website problem 😺.
that's what I did for failing routes in the frontend (see https://github.com/serlo/frontend/issues/1254) and it was really  worth it.
Your call.

File: src/embed.ts (hunk 2 above)
Comment (elbotho) [reply src/embed.ts:137]: well okay, we'll figure it out if there is a problem with this regex at some point ;)

**Comment by kulla** [comment]
> All in all it seems very exact a lot of code,

Yes, we can shorten the code with using the TaskEither monad though :smile: Might be something for the future...

> have there been a lot of problems in this module or do you want to archive this level of reporting everywhere in the long run?

No, it is only necessary in `src/embed` since there a lot of places where we use `getPlaceholder()`. Many of those cases are interesing for us (e.g. an API has a breaking change). However with `getPlaceholder()` we wouldn't notice it since there is no error thrown / message in sentry. Now we can be sure that everything is fine :smile:
